pkg/crc/preflight/preflight_checks_tray_windows.go

```diff
@@ -19,126 +18,139 @@ import (
+		return fmt.Errorf("Error trying to create symlink to tray in start-up folder: %w", err)
 	}
-
-	// invoke the ps script
-	_, _, err = powershell.ExecuteAsAdmin("Installing System Tray for CodeReady Containers", psFilePath)
-	// wait for the script to finish executing
-	time.Sleep(2 * time.Second)
-	if err != nil {
-		logging.Debug("Failed to execute tray installation script")
-		return err
+	cmd = fmt.Sprintf(`Start-Process -FilePath "%s"`, constants.TrayExecutablePath)
+	if _, _, err := powershell.Execute(cmd); err != nil {
+		return fmt.Errorf("Failed to start tray process: %w", err)
 	}
+	return nil
+}
 
-	// check for 'success' file
-	if _, err = goos.Stat(filepath.Join(tempDir, "success")); goos.IsNotExist(err) {
-		return fmt.Errorf("Installation script didn't execute successfully: %v", err)
+func removeTray() error {
+	_ = stopTray()
+	return goos.Remove(filepath.Join(constants.StartupFolder, constants.TrayShortcutName))
+}
+
+func stopTray() error {
+	trayProcessName := constants.TrayExecutableName[:len(constants.TrayExecutableName)-4]
+	cmd := fmt.Sprintf(`Stop-Process -Name "%s"`, trayProcessName)
+	if _, _, err := powershell.Execute(cmd); err != nil {
+		return fmt.Errorf("Failed to kill tray process: %w", err)
 	}
 	return nil
 }
 
-func escapeWindowsPassword(password string) string {
-	// escape specials characters (|`|$|"|') with '`' if present in password
-	if strings.Contains(password, "`") {
-		password = strings.ReplaceAll(password, "`", "``")
+func removeDaemon() error {
+	_ = stopDaemon()
+	var mErr errors.MultiError
+	if err := goos.Remove(constants.DaemonBatchFilePath); err != nil {
+		mErr.Collect(err)
 	}
-	if strings.Contains(password, "$") {
-		password = strings.ReplaceAll(password, "$", "`$")
+	if err := goos.Remove(constants.DaemonPSScriptPath); err != nil {
+		mErr.Collect(err)
 	}
-	if strings.Contains(password, "\"") {
-		password = strings.ReplaceAll(password, "\"", "`\"")
-	}
-	if strings.Contains(password, "'") {
-		password = strings.ReplaceAll(password, "'", "`'")
+	if len(mErr.Errors) == 0 {
+		return nil
 	}
-	return password
+	return mErr
 }
 
-func removeTray() error {
-	trayProcessName := constants.TrayExecutableName[:len(constants.TrayExecutableName)-4]
-
-	tempDir, err := ioutil.TempDir("", "crc")
+func stopDaemon() error {
+	executablePath, err := goos.Executable()
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to find current executable's path: %w", err)
 	}
-	defer func() {
-		_ = goos.RemoveAll(tempDir)
-	}()
-
-	psScriptContent := genTrayRemovalScript(
-		trayProcessName,
-		constants.TrayShortcutName,
-		constants.DaemonServiceName,
-		tempDir,
+	// get the PID of the process running command 'crc daemon --log-level debug'
+	getCrcProcessCmd := `(Get-WmiObject Win32_Process -Filter "name = 'crc.exe'")`
+	cmd := fmt.Sprintf(`(%s | Where-Object -Property CommandLine -Eq '"%s" daemon --log-level debug').ProcessId`,
```

Comment:
Hardcoding `daemon --log-level` here is fragile, this is going to silently break the day we change the command line used to spawn the daemon

Reply:
Added https://github.com/code-ready/crc/pull/1781/commits/8692885447bf55a292c94eb0cf59ec95574845e7 to handle this!